src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/Assert.cs

```diff
@@ -344,6 +345,83 @@ public static void FileDoesNotExist(MSBuildResult result, params string[] paths)
             }
         }
 
+        public static Stream ContainsEmbeddedResource(string assemblyPath, string resourceName)
+        {
+            var stream = ExtractManifest(assemblyPath, resourceName);
+            Assert.NotNull(stream);
+
+            return stream;
+        }
+
+        public static void DoesNotContainEmbeddedResource(string assemblyPath, string resourceName)
+        {
+            var stream = ExtractManifest(assemblyPath, resourceName);
+            Assert.Null(stream);
+        }
+
+        private static unsafe Stream ExtractManifest(string path, string expectedResourceName)
```

Comment:
😻

Reply:
This code doesn't work super well in Mac OS debug builds (not sure why) and causes the test to be flaky. I tried reproing it on a Mac and wasn't able to.  Seems from what happens when it fails that the stream is pointing out to the wrong memory area, but I haven't been able to get a repro assembly.

I plan to disable this "more exhaustive" check on mac OS and simply check that there is a resource embedded, but skip checking the contents.

I'll give it a try to produce a faulty assembly on my mac and dig a bit, but I don't plan to spend more than 1h on it. (Just enough to discard an actual product bug).

Reply:
I got a dll that caused issues, I'm going to debug the code and see if its a math mistake in the snippet, but if I can't find it I'll work something else out.

Reply:
I did change the snippet for something alternative and easier and filed a bug in corefx. Waiting to see if the build passes

Reply:
It did